Open Judge System/OJS.Common/Extensions/DirectoryHelpers.cs

```diff
@@ -20,6 +22,20 @@ public static string CreateTempDirectory()
             }
         }
 
+        public static string CreateTempDirectoryForExecutionStrategy()
+        {
+            while (true)
```

Comment:
I might say I am a little concerned about while(true)

Reply:
It's a measure against the randomness of the Path.GetRandomFileName() as it has (an incredibly low but still existent) chance to create a path that already exists.